tests/unit/test_hw_tools.py

```diff
@@ -764,6 +765,24 @@ def test_check(self, mock_check_deb_method):
         mock_check_deb_method.assert_called_with("smartmontools")
 
 
+@mock.patch("platform.machine")
+def test_get_release_url(mock_machine):
+    base_url = "https://example.com/releases/"
+    releases = {
+        "x86_64": "example-0.1.0.linux-amd64.tar.gz",
+        "arm64": "example-0.1.0.linux-arm64.tar.gz",
+    }
+
+    # Test for unsupported architecture
+    mock_machine.return_value = "unsupported_arch"
+    try:
+        get_release_url(base_url, releases)
+    except ValueError as e:
+        assert str(e) == "Unsupported architecture: unsupported_arch"
+    else:
+        assert False, "ValueError not raised"
```

Comment:
Pytest has a [mechanism for testing this pattern](https://docs.pytest.org/en/stable/reference/reference.html#pytest-raises):

```suggestion
    with pytest.raises(ValueError, match="Unsupported architecture: unsupported_arch"):
        get_release_url(base_url, releases)
```

:)